src/main/scala/org/hammerlab/guacamole/reads/ReadInputFilters.scala

```diff
@@ -0,0 +1,78 @@
+package org.hammerlab.guacamole.reads
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.SequenceDictionary
+import org.hammerlab.guacamole.loci.LociSet
+
+/**
+ * Filtering reads while they are loaded can be an important optimization.
+ *
+ * These fields are commonly used filters. For boolean fields, setting a field to true will result in filtering on
+ * that field. The result is the intersection of the filters (i.e. reads must satisfy ALL filters).
+ *
+ * @param overlapsLoci if not None, include only mapped reads that overlap the given loci
+ * @param nonDuplicate include only reads that do not have the duplicate bit set
+ * @param passedVendorQualityChecks include only reads that do not have the failedVendorQualityChecks bit set
+ * @param isPaired include only reads are paired-end reads
+ */
+case class ReadInputFilters(overlapsLoci: Option[LociSet.Builder],
+                            nonDuplicate: Boolean,
+                            passedVendorQualityChecks: Boolean,
+                            isPaired: Boolean)
+
+object ReadInputFilters {
+  val empty = ReadInputFilters()
+
+  /**
+   * See ReadInputFilters for full documentation.
+   *
+   * @param mapped include only mapped reads. Convenience argument that is equivalent to specifying all sites in
+   *               overlapsLoci.
+   */
+  def apply(mapped: Boolean = false,
+            overlapsLoci: LociSet.Builder = null,
```

Comment:
I wanted to support the interface `ReadInputFilters(overlapsLoci = loci)` instead of ``ReadInputFilters(overlapsLoci = Some(loci))`, so it's just a quick pass through null/Any space instead of `Option`-space